amethyst_input/src/input_handler.rs

```diff
@@ -628,3 +628,514 @@ where
+        sets_equal(
+            &event_vec,
+            &[
+                InputEvent::ActionReleased(String::from("test_combo_action")),
+                InputEvent::KeyReleased {
+                    key_code: VirtualKeyCode::Up,
+                    scancode: 104,
+                },
+                InputEvent::ButtonReleased(Button::Key(VirtualKeyCode::Up)),
+                InputEvent::ButtonReleased(Button::ScanCode(104)),
+            ],
+        );
+
+        handler.send_event(
+            &Event::WindowEvent {
+                window_id: unsafe { WindowId::dummy() },
+                event: WindowEvent::KeyboardInput {
+                    device_id: unsafe { DeviceId::dummy() },
+                    input: KeyboardInput {
+                        scancode: 112,
+                        state: ElementState::Released,
+                        virtual_keycode: Some(VirtualKeyCode::Down),
+                        modifiers: ModifiersState {
+                            shift: false,
+                            ctrl: false,
+                            alt: false,
+                            logo: false,
+                        },
+                    },
+                },
+            },
+            &mut events,
+            1.0,
+        );
+        assert!(!handler.action_is_down("test_combo_action").unwrap());
+        let event_vec = events.read(&mut reader).cloned().collect::<Vec<_>>();
+        sets_equal(
+            &event_vec,
+            &[
+                InputEvent::KeyReleased {
+                    key_code: VirtualKeyCode::Down,
+                    scancode: 112,
+                },
+                InputEvent::ButtonReleased(Button::Key(VirtualKeyCode::Down)),
+                InputEvent::ButtonReleased(Button::ScanCode(112)),
+            ],
+        );
+    }
+
+    #[test]
+    fn emulated_axis_response() {
+        // Register an axis triggered by two keys
+        // Check that with nothing pressed we return 0.
+        // Press the positive and check for a positive response
+        // Release the positive, press the negative and check for a negative respones
+        // Press both and check for 0.
+        // Release both and check for 0.
+
+        let mut handler = InputHandler::<String, String>::new();
+        let mut events = EventChannel::<InputEvent<String>>::new();
+        handler.bindings.insert_axis(
+            String::from("test_axis"),
+            Axis::Emulated {
+                pos: Button::Key(VirtualKeyCode::Up),
+                neg: Button::Key(VirtualKeyCode::Down),
+            },
+        );
+        assert_eq!(handler.axis_value("test_axis"), Some(0.0));
+        handler.send_event(
+            &Event::WindowEvent {
+                window_id: unsafe { WindowId::dummy() },
+                event: WindowEvent::KeyboardInput {
+                    device_id: unsafe { DeviceId::dummy() },
+                    input: KeyboardInput {
+                        scancode: 112,
```

Comment:
do you want to make this `104` so as to not overlap with the `Down` key? same in line 1037, 1081, and 1103